Source/PreferencesWindowController.swift

```diff
@@ -178,6 +185,68 @@ fileprivate let kWindowTitleHeight: CGFloat = 78
         }
     }
 
+    func updateUserPhraseLocation() {
+        if Preferences.useCustomUserPhraseLocation {
+            userPhrasesTextField.stringValue = Preferences.customUserPhraseLocation ?? ""
+            openUserPhrasesFolderButton.title = Preferences.customUserPhraseLocation ?? ""
+        } else {
+            userPhrasesTextField.stringValue = ""
+            openUserPhrasesFolderButton.title = UserPhraseLocationHelper.defaultUserPhraseLocation
+        }
+    }
+
+    @IBAction func changeUserPhraseEnabledAction(_ sender: Any) {
+        guard let control = sender as? NSControl else {
+            return
+        }
+        let enabled = control.integerValue > 0
+        if enabled {
+            if Preferences.customUserPhraseLocation == nil || Preferences.customUserPhraseLocation == "" {
+                Preferences.customUserPhraseLocation = UserPhraseLocationHelper.defaultUserPhraseLocation
+            } else {
+                Preferences.customUserPhraseLocation = Preferences.customUserPhraseLocation
```

Comment:
Just to confirm: This is setting the property to its current value. Is this to trigger the posting of  the `.customUserPhraseLocationDidChange` notification?